features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryServiceRegistryImpl.java

```diff
@@ -121,7 +121,13 @@ public T getService(BD beanDefinition) {
             final T service = (T) registration.getServiceFactory().createBean(beanDefinition);
 
             final BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(service);
-            wrapper.setPropertyValues(beanDefinition.getParameterMap());
+//            wrapper.setPropertyValues(beanDefinition.getParameterMap());
+            // TODO fooker: Workaround for unknown properties in BmpIntegrationAdapter (kafka.*)
```

Comment:
This seems like a reasonable fix -  only set properties that are writeable - caveat is that it makes it harder to detect mis-configured properties.

If the property is not writeable, maybe we could call some known function on the bean to validate?

Reply:
Deferred to https://issues.opennms.org/browse/NMS-12573